grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/HttpStreamDeframer.java

```diff
@@ -146,6 +151,17 @@ public void processOnError(Throwable cause) {
                 GrpcStatus.fromThrowable(statusFunction, ctx, cause, metadata), metadata);
     }
 
+    @Override
+    public void processOnComplete(StreamDecoderInput in, StreamDecoderOutput<DeframedMessage> out)
+            throws Exception {
+        if (ctx instanceof ClientRequestContext && !trailersReceived) {
```

Comment:
How about adding a `boolean server` field instead?

Reply:
Sounds good. It was another idea while implementing this.